utils/epinow2/functions.py

```diff
@@ -0,0 +1,140 @@
+from datetime import datetime
+from uuid import uuid1
+
+from utils.epinow2.constants import (
+    nssp_states_omit,
+    pathogens,
+    shared_params,
+    states,
+)
+
+
+def generate_job_id():
+    """Generates a UUID1 object to associate
+    with job IDs. UUID1 can be sorted by
+    timestamp by default, which is desirable
+    when managing multiple configurations."""
+    return uuid1()
+
+
+def validate_args(
+    state=None,
```

Comment:
```suggestion
    state: str | None = None,
```
If possible, I think we should use type hints here to help with readability. For cases like this where it could be some type or `None`, we can either write `<some_type> | None` or `typing.Optional[<some_type>]`. Personally, I prefer the first. That said, the first does require at least python 3.10, but I don't think that will be much of an issue